content/terraform/v1.13.x/docs/language/meta-arguments/lifecycle.mdx

```diff
@@ -0,0 +1,320 @@
+---
+page_title: lifecycle meta-argument reference
+description: How to configure the lifecycle meta-argument rules in Terraform--create_before_destroy, prevent_destroy, ignore_changes, replace_triggered_by, precondition, postcondition, destroy
+---
+
+# `lifecycle` reference
+ 
+Terraform performs the following operations when you apply a configuration:
+
+1. Creates resources defined in the configuration that are not associated with a real infrastructure object in the state.
+1. Destroys resources that exist in the state but not in the configuration.
+1. Updates resources in-place whose arguments have changed.
+1. Destroys and re-create resources whose arguments have changed but that Terraform cannot update in-place because of remote API limitations.
+
+The `lifecycle` block accepts a rule that customizes how Terraform performs the lifecycle stages for each resource. 
+
+## Usage
+
+All `lifecycle` settings affect how Terraform constructs and traverses
+the dependency graph. As a result, only literal values can be used because
+the processing happens too early for arbitrary expression evaluation.
+
+### `create_before_destroy` 
+
+By default, when Terraform must change a resource argument that cannot be updated in-place due to remote API limitations, Terraform destroys the existing object and then create a new replacement object with the new configured arguments. Use the `create_before_destroy` rule to instruct Terraform to create a replacement resource before destroying the current resource. 
+
+This is an opt-in behavior because many remote object types have unique name requirements or other constraints that must be accommodated for both a new and an old object to exist concurrently. Some resource types offer special options to append a random suffix onto each object name to avoid collisions, for example. Terraform CLI cannot automatically activate such features, so you must understand the constraints for each resource type before using `create_before_destroy` with it.
+
+Note that Terraform propagates and applies the `create_before_destroy` meta-attribute behavior to all resource dependencies. For example, if `create_before_destroy` is enabled on resource `A` but not on resource `B`, but resource `A` is dependent on resource `B`, then Terraform enables `create_before_destroy` for resource `B` implicitly by default and stores it to the state file. You cannot override `create_before_destroy`
+to `false` on resource `B` because that would imply dependency cycles in the graph.
+
+Destroy provisioners of this resource do not run if `create_before_destroy`
+is set to `true`. This [GitHub issue](https://github.com/hashicorp/terraform/issues/13549) contains more details.
```

Comment:
```suggestion
When the resource contains a provisioner that runs during the `destroy` operation,  setting `create_before_destroy` to `true` also prevents the provisioner from running. 
```
@BrianMMcClain - I agree, but I wanted to get your take on the original language, which IMO is kind of hard to follow if you don't have the concept of "destroy provisioner". Is this edit clearer?